### ISSUE TITLE
Add WhatsApp sender/recipient info to chat preview and metadata (for 2 parties chats)

### ISSUE THREAD

**Issue by mgd1980** [sepinf-inc/IPED#428]
In the IPED-Search preview windows, add the user data (ie name and phone number) related to that chat to the chat title. This information is necessary when we have cases with more than one Whatsapp account or iPhone backups. 

phoneParsersToUse = internal

**Comment by leosol** [comment]
Is anybody working on this?

**Comment by lfcnassif** [comment]
I think not. I would not add this info to chat title (that could be large), but instead change the report html to show sender name and phone for each message, like is done for group chats, and add the parties to ExtraProperties.PARTICIPANTS metadata.

**Comment by leosol** [comment]
Understood and seems simple this way. I did some coding and attached a picture of the results.

 ```
               //ReportGenerator.java (printMessage) (line 280 and below)
                String name = null, number = null;
                if (message.isFromMe()) {
                    out.println("<div class=\"outgoing to\">"); //$NON-NLS-1$
                    if(account!=null) {
                    	name = account.getName();
                    	number = message.getLocalResource();
                    } else {
                    	name = null;
                    	number = message.getLocalResource();
                    }
                } else {
                    out.println("<div class=\"incoming from\">"); //$NON-NLS-1$
                    number = message.getRemoteResource();
                    if (number != null) {
                        WAContact contact = contactsDirectory.getContact(number);
                        name = contact == null ? null : contact.getName();
                    }
                }

                //WhatsAppParser->createReport (line 227)
                if (account != null) {
	                String local = formatContact(account, cache);
	                chatMetadata.add(ExtraProperties.PARTICIPANTS, local);
                }
                if(c.isGroupChat()) {
	                for(WAContact member:c.getGroupmembers()) {
	                	chatMetadata.add(ExtraProperties.PARTICIPANTS, formatContact(member, cache));
	                }
                } else {
	                if(c.getRemote()!=null) {
	                	chatMetadata.add(ExtraProperties.PARTICIPANTS, formatContact(c.getRemote(), cache));
	                }
                }
```

![ISSUE-428-sender-recipient](https://user-images.githubusercontent.com/4167122/115973945-2f448e00-a52f-11eb-91c3-3344dd21e126.png)

**Comment by lfcnassif** [comment]
Great, thank you! Will take a look possibly on Monday.

**Comment by lfcnassif** [comment]
Did a quick reading (no testing yet), this possibly works. A PR or patch would be better to see removed or added lines, thanks.

**Comment by lfcnassif** [comment]
closed by #495
